### PR TITLE
Use mean as reduction method for loss with DataParallel training

### PR DESCRIPTION
**Pull request by pyeres**
This change makes loss metrics under different GPU configs more consistent by computing the average loss (instead of sum) across GPUs:
* 4b920c6 adds 'mean' as a reduction type option to `get_output_attribute` (the fn used to reduce scalars gathered in the output dict during DataParallel training). 
* 07a3b15 switches the `loss` calculation under DataParallel training to use 'mean' (instead of 'sum') as its reduction fn.

### CODE REVIEW

**Comment by sleepinyourhat** [comment]
Mind pinging others for a review? Otherwise can take a look by Mon.

**Comment by pyeres** [comment]
@pruksmhc and @zphang — you're probably the most familiar with this code. Can you review?

**Comment by pruksmhc** [comment]
Are you sure that without DataParallel we're getting the average loss per batch? Later on in trainer.py we divide the loss by the batch_num as well, so worth double checking.

**Comment by pyeres** [comment]
Thanks for the comments, @pruksmhc — here's why I think we want to use mean:

> Are you sure that without DataParallel we're getting the average loss per batch?

My read is that [pytorch loss fns](https://pytorch.org/docs/stable/nn.html#torch.nn.CrossEntropyLoss) use 'mean' as their default reduction method and I don't see jiant code overriding that. So without DataParallel I'd expect [this loss](https://github.com/nyu-mll/jiant/blob/07a3b15f078f82607dd746e29f4c0b69a16f1063/jiant/trainer.py#L583-L587) to be avg loss per batch.

> Later on in trainer.py we divide the loss by the batch_num as well, so worth double checking.

The division you're referring to might be [this cumulative loss](https://github.com/nyu-mll/jiant/blob/f148485acf438378f430ae1dae92c4d56363a243/jiant/trainer.py#L591) in `task_info` [later divided by steps](https://github.com/nyu-mll/jiant/blob/07a3b15f078f82607dd746e29f4c0b69a16f1063/jiant/trainer.py#L613) (before a6fef61 that cumulative loss in `task_info` was just called `loss` which was confusing).

I also did a quick check that training losses were consistent using `mean` reduction on training runs with different numbers of GPUs (and different settings of `accumulation_steps`).

**Comment by pyeres** [comment]
Thanks for the review, @pruksmhc — merging.
